drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb

```diff
@@ -600,105 +550,180 @@ def self.question_number
         }
       end
 
-      # Table 6c.1
-      # Metric 6c.1 mirrors Measure 7b.1 but with a universe of Category 3 SH, TH, PH – RRH, and PH (without a housing move-in date) (Project types: 8, 2, 13, and 3 without a move-in-date) system leavers in the current reporting year.
-      # Table 7b.1
-      # 1. Select leavers across all ES, SH, TH, PH-RRH, and PH projects in
-      # the in the report date range. A “leaver” in this metric means the
-      # client must have exited from a project of one of the given types in
-      # the report date range and was not active in that or any other project
-      # among the given types as of the [report end date].
-      m7b_excluded_destinations = [15, 6, 25, 24]
-      m7b1_project_types = ES + SH + TH + PH
-      m7b1_exits = add_filters m7_exits.
-        hud_project_type(
-          m7b1_project_types,
-        ).where.not(
-          client_id: m7_stays.hud_project_type(m7b1_project_types).select(:client_id),
-        )
+      add_project_leavers_and_stayers 'Measure 7', :m7
+    end
+
+    # #Metric 7b and  Metric 6c are mostly the same but
+    private def add_project_leavers_and_stayers(question_name, m_code)
+      return unless add_clients_for_question?(question_name)
+
+      stays = add_filters GrdaWarehouse::ServiceHistoryEnrollment.entry.ongoing(
+        on_date: @report.end_date,
+      )
 
-      process_scope_by_client(question_name, m7b1_exits, m7_columns) do |_client, client_enrollments|
+      exits = add_filters GrdaWarehouse::ServiceHistoryEnrollment.exit.ended_between(
+        start_date: @report.start_date,
+        end_date: @report.end_date + 1.day,
+      )
+
+      if m_code == :m6
+        history_field = :m6_history
+        table_1_dest_field = :m6c1_destination
+        table_2_dest_field = :m6c2_destination
+
+        # Metric 6c.1 mirrors Measure 7b.1 but with a universe of Category 3
+        #  Safe Haven (project type 8),
+        #  Transitional Housing (Project Type 2),
+        #  Permanent Housing-Rapid Rehousing (Project Type 13),
+        #  and PH-PSH (Project Type 3, without a housing move-in date) clients who exited during the report range.
+        # Metric 6c.2: mirrors Measure 7b.2 but with a universe of Category 3
+        #  Permanent Housing – Permanent Supportive Housing (Project Type 3, with a housing move-in date) clients who are active in the report range.
+        exits = exits.category_3
+        table_1_project_types = [2, 8, 13, 3]
+        table_2_project_types = [3]
```

Comment:
This is fine, since the spec defines these explicitly, but why use integers here and constants on line 591-592?

Reply:
because the spec in this case used specific numbers and i composing them from named constants was getting confusing.